modules/oe_webtools_analytics/modules/oe_webtools_analytics_rules/src/EventSubscriber/WebtoolsAnalyticsEventSubscriber.php

```diff
@@ -96,16 +126,34 @@ public function analyticsEventHandler(AnalyticsEventInterface $event): void {
     $rules = $storage->loadMultiple();
     /** @var \Drupal\oe_webtools_analytics_rules\Entity\WebtoolsAnalyticsRuleInterface $rule */
     foreach ($rules as $rule) {
-      if (preg_match($rule->getRegex(), $current_uri, $matches) === 1) {
-        $event->setSiteSection($rule->getSection());
-        $this->cache->set($current_uri, ['section' => $rule->getSection()], Cache::PERMANENT, $rule->getCacheTags());
-        // Currently there is no defined behavior for overlapping rules so we
-        // only take into account the first rule that applies.
-        return;
+      if ($rule->isSupportMultilingualAliases()) {
+        // Get source of current URI.
+        // But some reason we don't have correct information about current path.
+        // For updating information
+        // we have to run $this->routeProvider->getRouteCollectionForRequest().
+        $this->routeProvider->getRouteCollectionForRequest($this->requestStack->getCurrentRequest());
+        $source = \Drupal::service('path.current')->getPath();
+
+        $query = \Drupal::database()->select('url_alias', 'ua');
+        $query->fields('ua', ['pid']);
+        $query->condition('ua.source', $source);
+        // As mysql doesn't support PCRE, we have to remove modifiers.
+        $regexp = preg_replace(['/^\//', '/\/.?$/'], '', $rule->getRegex());
+        $query->condition('ua.alias', $regexp, 'REGEXP');
+        $default_langcode = \Drupal::config('system.site')->get('default_langcode');
```

Comment:
Shouldn't we inject all services above? Meaning ` \Drupal::config()`, `\Drupal::database()`, `\Drupal::service('path.current')`.

Reply:
Unfortunately the AliasStorage does not support regex search but still we should base this lookup on `\Drupal\Core\Path\AliasStorage::lookupPathAlias()`, especially since that method adds `LanguageInterface::LANGCODE_NOT_SPECIFIED` to the list of languages:

```php
  /**
   * {@inheritdoc}
   */
  public function lookupPathAlias($path, $langcode) {
    $source = $this->connection->escapeLike($path);
    $langcode_list = [$langcode, LanguageInterface::LANGCODE_NOT_SPECIFIED];
```

We should make sure if that's relevant for our case. I would isolate this into a method anyway and write a kernel test that tests it under different conditions.

Reply:
In initial functionality we could specify regular expression which could contain any complexity error rules which could be not be transferred to mysql LIKE

Reply:
Fixed